catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/stage/ParticleEffectsTest.kt

```diff
@@ -0,0 +1,194 @@
+        Assert.assertFalse(projectManager.currentSprite.look.isParticleEffectPaused)
+    }
+
+    @Test
+    fun particleEffectGravityTest() {
+        script.addBrick(FadeParticleEffectBrick(0))
+        script.addBrick(WaitBrick(400))
+        script.addBrick(SetGravityBrick(Formula(0.0), Formula(20.0)))
+        Espresso.onView(ViewMatchers.withId(R.id.button_play)).perform(ViewActions.click())
+        val projectManager = projectManager
+        Espresso.onView(ViewMatchers.isRoot()).perform(CustomActions.wait(100))
+        val look = projectManager.currentSprite.look
+        val emitter = look.particleEffect.emitters[0]
+        var particleGravity = emitter.gravity.highMax.toDouble()
+        Assert.assertEquals(particleGravity, -10.0, 0.00)
+        Espresso.onView(ViewMatchers.isRoot()).perform(CustomActions.wait(500))
+        particleGravity = emitter.gravity.highMax.toDouble()
+        Assert.assertEquals(particleGravity, 20.0, 0.0)
+    }
+
+    @Test
+    fun particleEffectOnCloneTest() {
+        script.addBrick(FadeParticleEffectBrick(0))
+        script.addBrick(CloneBrick())
+        Espresso.onView(ViewMatchers.withId(R.id.button_play)).perform(ViewActions.click())
+        val sprites = StageActivity.stageListener.spritesFromStage
+        var cloneFound = false
+        for (sprite in sprites) {
+            if (sprite.isClone) {
+                cloneFound = true
+                Assert.assertTrue(sprite.look.hasParticleEffect)
+            }
+        }
+        if (!cloneFound) {
+            Assert.fail("No Clone Found")
+        }
+    }
+
+    @Test
+    fun particleAfterWaitTest() {
+        script.addBrick(WaitBrick(2000))
+        script.addBrick(FadeParticleEffectBrick(0))
+        Espresso.onView(ViewMatchers.withId(R.id.button_play)).perform(ViewActions.click())
+        val projectManager = projectManager
+        val look = projectManager.currentSprite.look
+        Assert.assertFalse(look.hasParticleEffect)
+        Espresso.onView(ViewMatchers.isRoot()).perform(CustomActions.wait(2000))
+        Assert.assertTrue(look.hasParticleEffect)
+    }
+
+    @Test
+    fun particleEffectDisableAdditiveByDefaultWhenNoBackgroundLookTest() {
+        script.addBrick(FadeParticleEffectBrick(0))
+        Espresso.onView(ViewMatchers.withId(R.id.button_play)).perform(ViewActions.click())
+        Assert.assertFalse(sprite.look.isAdditive)
+    }
+
+    @Test
+    fun particleEffectEnableAdditiveByDefaultWhenNoBackgroundLookTest() {
+        projectManager.currentlyEditedScene.backgroundSprite.lookList.add(LookData())
+        script.addBrick(FadeParticleEffectBrick(0))
+        Espresso.onView(ViewMatchers.withId(R.id.button_play)).perform(ViewActions.click())
+        Assert.assertTrue(sprite.look.isAdditive)
+    }
+
+    @After
+    fun tearDown() {
+        baseActivityTestRule.finishActivity()
+        try {
+            StorageOperations.deleteDir(File(FlavoredConstants.DEFAULT_ROOT_DIRECTORY, projectName))
```

Comment:
no need for that, the test rule will take care of that for you.
for more info see `BaseActivityTestRule.java`